gnosis/safe-react-apps
# Feature/custom contract data

## What it solves
Resolves #214 

## How this PR fixes it
We are adding a Switch for allowing the user to introduce custom data

When the user activates the switch then a `textarea` is presented. If the user was previously selected a method and filled the individual inputs and the validation result is positive then we are prefilling the textarea with the custom data.

Adding as well a "Show more/less" link in the preview modal before sending the transactions to the safe-web

We can send some Native token amount with the custom transaction data

## How to test it
1) Open TX Builder
2) <ins>Case 1</ins>. Select a method, fill the inputs with correct data and before adding the transaction toggle the switch => The custom data should be prefilled. Test the reverse case where the user is prefilling something wrong and then the textarea is not prefilled. Add the transaction
3) <ins>Case 2</ins>. Toggle the switch and enter the custom data. Should be hex strict or an error will be showed. Add transaction
4) <ins>Case 3</ins>. Add custom data with a length greater than 50 characters. In the popup with the transactions resume, the custom data with a "Show more/less" link should be showed
5) <ins>Case 4</ins>. Send some native token amount with the custom data and check is reflected in the final safe-web transaction

## Screenshots

![image](https://user-images.githubusercontent.com/1576776/145071679-58c7b053-24e9-4db7-8163-5e072ce26e75.png)

![image](https://user-images.githubusercontent.com/1576776/145071709-3de876f0-2d3e-4eab-bffb-e617509f7c75.png)
